python/dgl/graphbolt/impl/ondisk_dataset.py

```diff
@@ -79,66 +83,82 @@ def preprocess_ondisk_dataset(
     os.makedirs(os.path.join(dataset_dir, processed_dir_prefix), exist_ok=True)
     output_config = deepcopy(input_config)
 
-    # 2. Load the edge data and create a DGLGraph.
+    # 2. Load the edge data and create a FusedCSCSamplingGraph.
     if "graph" not in input_config:
         raise RuntimeError("Invalid config: does not contain graph field.")
     is_homogeneous = "type" not in input_config["graph"]["nodes"][0]
     if is_homogeneous:
         # Homogeneous graph.
-        num_nodes = input_config["graph"]["nodes"][0]["num"]
         edge_data = pd.read_csv(
             os.path.join(
                 dataset_dir, input_config["graph"]["edges"][0]["path"]
             ),
             names=["src", "dst"],
         )
-        src, dst = edge_data["src"].to_numpy(), edge_data["dst"].to_numpy()
-
-        g = dgl.graph((src, dst), num_nodes=num_nodes)
+        coo_tensor = torch.LongTensor(edge_data.values).T
```

Comment:
Do we need define a helper function here to avoid huge function body? So that we can directly **test** this helper function first.

Reply:
any suggestions for the location?

Reply:
+1

if this util function is for internal use only, it's ok to place in this file and the name starts with `_`.

Reply:
> any suggestions for the location?

I would like to add the function into `internal` folder and test it in the `test_internal.py` as the COO to CSC function may be used elsewhere?

Reply:
> I would like to add the function into `internal` folder and test it in the `test_internal.py` as the COO to CSC function may be used elsewhere?

The code that converts COO to CSC is not _huge_ at all, only three lines:

```python
    coo_tensor = torch.tensor(edge_data.values).T
    sparse_matrix = dglsp.spmatrix(coo_tensor)
    indptr, indices, value_indices = sparse_matrix.csc()
```

What makes the code so long is the parsing of yaml data.